src/Content/Text/BBCode.php

```diff
@@ -1361,6 +1361,15 @@ function ($match) {
 				. '</a>';
 		}, $text);
 
+		// We need no target="_blank" for local links
+		// convert links start with System::baseUrl() as local link
+		$escapedBaseUrl = str_replace('://', '\:\/\/', System::baseUrl());
+		$text = preg_replace("/\[url\]($escapedBaseUrl)([$URLSearchString]*)\[\/url\]/ism", '<a href="$1$2">$1$2</a>', $text);
+		$text = preg_replace("/\[url\=($escapedBaseUrl)([$URLSearchString]*)\](.*?)\[\/url\]/ism", '<a href="$1$2">$3</a>', $text);		
+		// convert links that start with / as local link
+		$text = preg_replace("/\[url\](\/[$URLSearchString]*)\[\/url\]/ism", '<a href="'.System::baseUrl().'$1">$1</a>', $text);
```

Comment:
We don’t support local links in BBCode text, probably for security reasons. Most texts are coming from remotes nodes, so local links make little sense. Please remove this.

Reply:
probably for security reasons? What reasons could that be? I can set a link with my host as domain. That link opens in a new tab. This is annoying and useless. 

In my opinion, the target="_blank" is useless. Shouldnt it be a decision by the user, if he wants to open a link in a new tab at all? Most people i know, really dont like this behaviour of opening tabs. 

When I browse a remote users profile on my host, and click on the username, the remote host opens in the same tab?! 

Imagine a user with a page type or forum type setup. The page/forum is used as kind about/info/support page/forum for that host. A lot of links to other local forumposts or pages could be daily business. Each of them opens a new tab .. to me this is bad user experience.

What do you think about this scenario?

Reply:
Security is more about what you cautiously allow than what you disallow. In this case, We don't have a good reason to allow relative links in posts because they're supposed to be distributed to remote nodes where these relative links may not make any sense (Diaspora sends relative links for posts and profiles and we had to have special provisions to handle them).

On the other hand, if someone crafted a malicious link, disallowing relative links means that it will only affect a singe node.

However, this has nothing to do with `taget="_blank"` . I firmly believe that people should be given the choice instead of forcing a behavior, but this isn't the consensus here at Friendica, and so far external links have been forced to be open in new tabs/window.

I like your approach of detecting local links and selectively removing `target="_blank"` but it doesn't have anything to do with my reluctance to allow relative links.

Reply:
Yeah, got your point. I didnt know that they are sent out as it. I was just testing it. ;)
So we could check the [url] before sending and decide how to handle local links. for outgoing they should add the target="_blank", on local host render without. Relative links should be completed with System::basUrl().

Reply:
Ideally, yes. in practice, the BBCode conversion should not need to know what is the purpose of the conversion, to limit complexity. As a result, we don't know if it's for sending or receiving. Prepending the Friendica base URL before Diaspora relative links doesn't make sense either.

Reply:
removed preg_replace for relative urls
done

Reply:
Thanks, please see the other comment for correct function name.